### ISSUE TITLE
A few questions

### ISSUE THREAD

**Issue by peterlewis** [maxgoedjen/secretive#111]
Sorry to be a pain, but just so I can understand, slightly more, are you able to help me answer a three Qs?

1. If 'User A' creates a key, what prevents 'User B' from being able to see this key within the secure enclave, given that the bundle ID is the same? I'm guessing this is keychain-related, but keen to understand?

2. When testing uninstall, I did a `brew zap secretive`, which removed all the files within the `~/Containers` directory. Had I not deleted the secret keys, first, would I have ended up orphaning anything?

3. Once removing any created keys, does anything remain within the secure enclave that was provisioned by 'Secretive'?

Basically, I'm just trying to understand how to avoid unintentionally orphaning keys, within the secure enclave, whilst I'm playing around! :)

Many thanks!

Peter

**Comment by martinh2011** [comment]
Would love to have answers to these in the FAQ ;-)

From what I see in the secretive source the keys are generated in a way that the user needs to unlock (e.g. via TouchID) before a key can be used. And with the secure enclave the access control is done by the enclave. So no other user should be able to use the keys.

I expect that the keys remain in the T2 chip if you delete the app. brew zap just deletes the app and a few folders in ~/Library. See https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/secretive.rb.

**Comment by maxgoedjen** [comment]
Hey @peterlewis – sorry for the delayed response, I missed this one.

1: https://support.apple.com/guide/security/keychain-data-protection-overview-secb0694df1a/web should answer some of your questions wrt users – this is a behavior of the keychain rather than the secure enclave itself.

2: I don't personally maintain the brew formula (a generous member of the community does) – but this lines up with my expectation of the behavior I'd have – there's no real state maintained by Secretive beyond the keychain, so I would not expect removing the ~/Containers directories to have any effect on key storage.

3: Not to my knowledge. I could be wrong on this, but I think everything should be cleaned out.

**Comment by maxgoedjen** [comment]
[Closing this for now, but please feel free to reopen if you have any followup questions 👍]
